src/features/mcda/components/MCDALayerEditor/MCDALayerParameters/TransformationsChart/TransformationsChart.tsx

```diff
@@ -0,0 +1,142 @@
+  transformedData,
+  originalData,
+}: TransformationsChartProps) {
+  const transformationApplied = useMemo(
+    () => transformedData?.transformation !== 'no',
+    [transformedData],
+  );
+
+  const clampedTransformedPoints = useMemo(
+    () =>
+      transformationApplied
+        ? transformedData?.points?.map((point) =>
+            inViewCalculations.clamp(
+              point,
+              transformedData?.lowerBound,
+              transformedData.upperBound,
+            ),
+          )
+        : transformedData?.points,
+    [transformationApplied, transformedData],
+  );
+
+  const data = useMemo(
+    () =>
+      originalData?.points?.map((value, index) => ({
+        transformed: clampedTransformedPoints?.[index],
+        original: value,
+        x: index,
+      })),
+    [clampedTransformedPoints, originalData?.points],
+  );
+  const scaleTransformed = useMemo(
+    () => getScaleFunction('no', clampedTransformedPoints),
+    [clampedTransformedPoints],
+  );
+  const scaleOriginal = useMemo(
+    () => getScaleFunction(transformedData?.transformation, originalData?.points),
+    [originalData?.points, transformedData?.transformation],
+  );
+
+  return (
+    <LineChart
+      width={400}
+      height={300}
+      data={data}
+      style={{ marginLeft: '-150px' }}
+      margin={{
+        top: 5,
+        right: 30,
+        left: 30,
+        bottom: 5,
+      }}
+    >
+      <CartesianGrid strokeDasharray="3 3" />
+      <XAxis
+        className={s.transformationChart}
+        dataKey={'x'}
+        ticks={[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]}
+        name="percentile"
+        label={{ value: 'Percentiles', position: 'insideBottomRight', offset: 0 }}
+      />
+      <YAxis dataKey="transformed" yAxisId="transformed" scale={scaleTransformed} />
+      <YAxis
+        dataKey="original"
+        scale={scaleOriginal}
+        orientation="right"
+        yAxisId="original"
+        axisLine={{ stroke: COLOR_ORIGINAL }}
+        tickLine={{ stroke: COLOR_ORIGINAL }}
+        tick={{ fill: COLOR_ORIGINAL }}
+      />
+      <Tooltip />
+      <Legend />
+      <Line
+        type="monotone"
+        dataKey="original"
+        stroke={COLOR_ORIGINAL}
+        yAxisId="original"
+        dot={{ r: 1.5, fill: COLOR_ORIGINAL }}
+        strokeWidth={0.25}
+        animationDuration={700}
+      />
+      <Line
+        type="monotone"
+        dataKey="transformed"
+        stroke={CHART_GREEN}
+        yAxisId="transformed"
+        dot={<GradientDot />}
+        strokeWidth={0.25}
+        animationDuration={700}
+      />
+    </LineChart>
+  );
+}
+
+const gradient = generateHclGradientColors(CHART_RED, CHART_YELLOW, 50);
+gradient.push(...generateHclGradientColors(CHART_YELLOW, CHART_GREEN, 51));
+
+const GradientDot = (props) => {
+  const { cx, cy, payload } = props;
+  return <circle cx={cx} cy={cy} r={1.5} fill={gradient[payload.x]} />;
+};
+
+TransformationsChart.propTypes = {};
```

Comment:
Why we need this?

Reply:
We don't. This code isn't even in the PR anymore